go.mod

```diff
@@ -0,0 +1,18 @@
+module github.com/census-instrumentation/opencensus-service
+
+require (
```

Comment:
Great question!
So the `go.mod` and `go.sum` files are just the manifests. We could not choose not check-in the entire `vendor/`* directory but that's going to require users of this repo to manually install it themselves. Go doesn't have an automatic integration to automatically build the vendored versions; unlike say `mvn install`. I'd vote for not including `vendor/` but just going with what was done before.

Hope this makes sense?

Reply:
@rakyll are you okay with us keeping go.mod and go.sum but then requiring users to manually use `go mod` and install their vendor directory themselves?

Reply:
How much burden do we add to the users of this repo if we keep "go.mod and go.sum" but no "vendor/*"?

Reply:
For now, since this is fresh work we won't have any burden for the new agent but those using @rakyll's agent will need the `vendor/`  directory which is why perhaps the old agent and new one need to live in separate repos.

Reply:
But if we remove the vendor/ in order to build what is currently in this repo what do I have to do? Do I just need to run a command and eveything is getting set or I have to do more things?

Reply:
>But if we remove the vendor/ in order to build what is currently in this repo what do I have to do? Do I just need to run a command and eveything is getting set or I have to do more things?

If we remove `vendor/` then you can definitely still build it, which is what we have after #40  before this PR is merged:
### Without `go.mod` or `go.sum` checked in:
`go get <package in here>` e.g. `go get github.com/census-instrumentation/opencensus-service/cmd/ocagent`

### With `go.mod` and `go.sum` checked in:
Like as per this PR but without `vendor/`
```shell
cd $(go env GOPATH)/src/github.com/census-instrumentation/opencensus-service && GO111MODULE=on go mod && go get github.com/census-instrumentation/opencensus-service/cmd/ocagent
```

Reply:
Make sense to me.